File: app/src/main/java/com/imangazalievm/bubbble/domain/interactors/UserShotsInteractor.java
package com.imangazalievm.bubbble.domain.interactors;


import com.imangazalievm.bubbble.domain.models.Shot;
import com.imangazalievm.bubbble.domain.models.User;
import com.imangazalievm.bubbble.domain.models.UserShotsRequestParams;
import com.imangazalievm.bubbble.domain.repository.IShotsRepository;
import com.imangazalievm.bubbble.domain.repository.IUsersRepository;

import java.util.List;

import javax.inject.Inject;

import io.reactivex.Single;

public class UserShotsInteractor {

    private IUsersRepository usersRepository;
    private IShotsRepository shotsRepository;

    @Inject
    public UserShotsInteractor(IUsersRepository usersRepository, IShotsRepository shotsRepository) {
        this.usersRepository = usersRepository;
        this.shotsRepository = shotsRepository;
    }

    public Single<User> getUser(long userId) {
        return usersRepository.getUser(userId);
    }

    public Single<List<Shot>> getUserShots(UserShotsRequestParams requestParams) {
        return shotsRepository.getUserShots(requestParams);
    }

}

File: app/src/main/java/com/bubbble/domain/userprofile/UserShotsInteractor.kt
package com.bubbble.domain.userprofile

import com.bubbble.data.shots.ShotsRepository
import com.bubbble.data.users.UsersRepository
import com.bubbble.models.Shot
import com.bubbble.models.User
import com.bubbble.models.UserShotsRequestParams
import javax.inject.Inject

class UserShotsInteractor @Inject constructor(
    private val usersRepository: UsersRepository,
    private val shotsRepository: ShotsRepository
) {

    suspend fun getUser(userId: Long): com.bubbble.models.User {
        return usersRepository.getUser(userId)
    }

    suspend fun getUserShots(requestParams: com.bubbble.models.UserShotsRequestParams): List<com.bubbble.models.Shot> {
        return shotsRepository.getUserShots(requestParams)
    }

}